students/Mintae_K/lesson04/mailroom2.py

```diff
@@ -0,0 +1,69 @@
+# Sending a Thank you
+def sendingthank(donors):
+    f_name = input('Please type the Full Name or list> ')
+    while f_name == 'list':
+        print(' ')
+        for x in donors:
+            print(x)
+        print(' ')
+        f_name = input('Please type the Full Name or list> ')
+    if donors.get(f_name, False):
+        s = 'Existing donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors[f_name].append(float(amount))
+    else:
+        s = 'New donor, please type the donation amount from {}> '
+        amount = input(s.format(f_name))
+        donors[f_name] = [float(amount)]
+    print('\n', "Thank you {} for the generous donation!".format(f_name), '\n')
+    return
+
+
+# Creating Report
+def createreport(donors):
+    s = '\n, {:<20}| {:<10} | {:<10}| {:<20}'
+    print(s.format('Donor Name', 'Total Given', "Num Gift", "Average Gift"))
+    print('-' * 60)
+    total = []
+    for keys in donors:
+        num_gift = len(donors[keys])
+        total.append([sum(donors[keys]), num_gift, keys])
+    total.sort(reverse=True)
+    s = '{:<20}  $ {:<10.2f}  {:^12} $ {:<20.2f}'
+    for i in range(len(total)):
+        print(s.format(total[i][2], total[i][0], total[i][1], total[i][0] / total[i][1]))
+    print(' ')
+    return
+
+
+def sendletters(donors):
+    for keys in donors:
```

Comment:
best to name variable that matches closely to what it holds - in this case it should be either singular `key` or `donor`, not plural